core/node/house_keeper/src/database.rs

```diff
@@ -0,0 +1,45 @@
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use zksync_dal::{system_dal::DatabaseMigration, ConnectionPool, Core, CoreDal};
+use zksync_health_check::{Health, HealthStatus, HealthUpdater};
+
+use crate::periodic_job::PeriodicJob;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DatabaseInfo {
+    last_migration: DatabaseMigration,
+}
+
+impl From<DatabaseInfo> for Health {
+    fn from(details: DatabaseInfo) -> Self {
+        Self::from(HealthStatus::Ready).with_details(details)
+    }
+}
+
+#[derive(Debug)]
+pub struct DatabaseHealthTask {
+    pub polling_interval_ms: u64,
+    pub connection_pool: ConnectionPool<Core>,
+    pub database_health_updater: HealthUpdater,
```

Comment:
Bikeshedding: To make encapsulation better, I'd suggest to make the fields private and use a constructor taking `connection_pool` and `polling_interval_ms` (a health check can be obtained via a getter, as with other components).

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/3193/commits/36ad7c5c2c4247510c52767f41f08ee7591d02cc and https://github.com/matter-labs/zksync-era/pull/3193/commits/1585d5dfbbe206e2d67a52c52d1b2910b244f409